From 73e5317de0f5bcdb8fc19a676f6f4226ed81b6bf Mon Sep 17 00:00:00 2001
From: Denis Barbier <bouzim@gmail.com>
Date: Fri, 29 Jul 2011 13:41:56 +0200
Subject: [PATCH] Unconditionnally include oce-config.h in inc/* header files

Applications which depend on OCE have still to pass
-DHAVE_CONFIG_H flag in order to compile, which is pretty
annoying.

Fixes https://github.com/tpaviot/oce/issues/15
---
 inc/OpenGl_FontMgr.hxx |    4 +---
 inc/Standard_Macro.hxx |    2 --
 inc/Standard_values.h  |   24 +++++++++++-------------
 inc/Xw_Extension.h     |    4 +---
 4 files changed, 13 insertions(+), 21 deletions(-)

Index: oce/inc/OpenGl_FontMgr.hxx
===================================================================
--- oce.orig/inc/OpenGl_FontMgr.hxx
+++ oce/inc/OpenGl_FontMgr.hxx
@@ -7,9 +7,7 @@
 # include <stdlib.h>
 #endif
 
-#ifdef HAVE_CONFIG_H
-# include <config.h>
-#endif
+#include <config.h>
 
 #ifdef HAVE_FTGL_NEWER212
 # include <FTGL/ftgl.h>
Index: oce/inc/Standard_Macro.hxx
===================================================================
--- oce.orig/inc/Standard_Macro.hxx
+++ oce/inc/Standard_Macro.hxx
@@ -7,9 +7,7 @@
 #ifndef _Standard_Macro_HeaderFile
 # define _Standard_Macro_HeaderFile
 
-#ifdef HAVE_CONFIG_H
 # include <config.h>
-#endif /* HAVE_CONFIG_H */
 
 // Standard OCC macros: Handle(), STANDARD_TYPE()
 # define   Handle(ClassName)  Handle_##ClassName
Index: oce/inc/Standard_values.h
===================================================================
--- oce.orig/inc/Standard_values.h
+++ oce/inc/Standard_values.h
@@ -22,11 +22,9 @@
 #error "Wrong compiler options has been detected. Add /DWNT option for proper compilation!!!!!"
 #endif
 
-#ifndef WNT
-#ifdef HAVE_CONFIG_H
-# include <config.h>
-#endif
+#include <config.h>
 
+#ifndef WNT
 #ifdef HAVE_LIMITS
 # include <limits>
 #elif defined (HAVE_LIMITS_H)
Index: oce/inc/Xw_Extension.h
===================================================================
--- oce.orig/inc/Xw_Extension.h
+++ oce/inc/Xw_Extension.h
@@ -22,9 +22,7 @@
 #define min(a,b) 	(a>b ? b : a)
 #endif
 
-#ifdef HAVE_CONFIG_H
-# include <config.h>
-#endif
+#include <config.h>
 
 #ifdef HAVE_X11_EXTENSIONS_TRANSOLV_H
 #include <X11/extensions/transovl.h>
